woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+    assert len(combined_df) == 8
+
+    assert combined_df.ww.time_index == 'signup_date'
+    assert combined_df.ww.schema == original_schema
+
+
+def test_concat_table_names(sample_df):
+    df1 = sample_df[['id', 'signup_date']]
+    df1.ww.init(name=0)
+    df2 = sample_df[['full_name', 'age']]
+    df2.ww.init(name='1')
+    df3 = sample_df[['phone_number', 'is_registered']]
+    df3.ww.init(name='2')
+
+    combined_df = concat([df1, df2, df3])
+    assert combined_df.ww.name == '0_1_2'
+
+    combined_df = concat([df3, df2, df1])
+    assert combined_df.ww.name == '2_1_0'
+
+
+def test_concat_cols_different_use_standard_tags(sample_df):
+    sample_df.ww.init(use_standard_tags={'age': False, 'full_name': False, 'id': True},
+                      logical_types={'full_name': 'Categorical'})
+
+    assert sample_df.ww.semantic_tags['id'] == {'numeric'}
+    assert sample_df.ww.semantic_tags['full_name'] == set()
+    assert sample_df.ww.semantic_tags['age'] == set()
+
+    df1 = sample_df.ww[['id', 'full_name', 'email']]
+    df2 = sample_df.ww[['phone_number', 'age', 'signup_date', 'is_registered']]
+
+    combined_df = concat([df1, df2])
+
+    assert combined_df.ww.semantic_tags['id'] == {'numeric'}
+    assert combined_df.ww.semantic_tags['full_name'] == set()
+    assert combined_df.ww.semantic_tags['age'] == set()
+
+
+def test_concat_rows_different_schema(sample_df):
+    df1 = sample_df.copy()
+    df1.ww.init(logical_types={'age': 'AgeNullable'})
+    df2 = sample_df.copy()
+    df2.ww.init(logical_types={'age': 'IntegerNullable'})
+    df3 = sample_df.copy()
+    df3.ww.init(logical_types={'age': 'NaturalLanguage'})
+
+    assert df2.ww.schema != df1.ww.schema
+
+    combined_df_1 = concat([df1, df2], axis=0)
+    combined_df_1.ww.logical_types['age'] == AgeNullable
+
+    combined_df_2 = concat([df2, df1], axis=0)
+    combined_df_2.ww.logical_types['age'] == IntegerNullable
+
+    combined_df_3 = concat([df3, df2], axis=0)
+    combined_df_3.ww.logical_types['age'] == NaturalLanguage
+    assert str(combined_df_3['age'].dtype) == 'string'
+
+
+def test_concat_cols_different_schema(sample_df):
+    df1 = sample_df[['id', 'age']]
+    df1.ww.init(index='id', logical_types={'id': 'AgeNullable'})
+    df2 = sample_df[['id', 'signup_date']]
+    df2.ww.init(index='id', logical_types={'id': 'IntegerNullable'})
+
+    assert df2.ww.schema != df1.ww.schema
+
+    combined_df_1 = concat([df1.ww.copy(), df2.ww.copy()], axis=1)
+    combined_df_1.ww.logical_types['id'] == AgeNullable
+
+    combined_df_2 = concat([df2.ww.copy(), df1.ww.copy()], axis=1)
+    combined_df_2.ww.logical_types['id'] == IntegerNullable
+
+
+def test_concat_rows_different_columns(sample_df):
+    sample_df.ww.init(time_index='signup_date')
+    df1 = sample_df.ww[['id', 'signup_date', 'full_name', 'email']]
+    df2 = sample_df.ww[['id', 'signup_date', 'phone_number', 'age', 'is_registered']]
+    # should work if dtypes dont change
```

Comment:
My mentality had mainly been that we let Woodwork initialize on the results of concat and if it fails, we let to woodwork initialization failure come through. So we'd get `TypeConversionError`s but it's also how we catch the duplicate column errors or any time the index column is suddenly not unique. It helps keep additional complexity out of the concat function.

I think if we try and stick with the comment above that the result of Woodwork's concat shouldn't make too many changes to the structure of the resulting dataframe (mostly coming from the handling of index and time index columns), then it could be less confusing about why woodwork initialization fails.